### DISCUSSION TITLE
Storage backup

### DISCUSSION THREAD

**Discussion by sdellava** [tobozo/ESP32-USB-Soft-Host#30] [Ideas]
Hi team!
I've an interesting use case for your library.  I've (and I'm one of many) a device that records video (mp4 files) on his internal memory. We can consider it a DVR. To save video, it is only possibile to connect to the device via USB and download the files to the pc. 

The Idea is to build a ESP32 based device where to connect the DVR. The ESP32, using your lib, have to continue to check the content of the DVR and when it find a new file, it  move on a connected SD memory.

Do you beleive it is possible to build such device?

**Comment by tobozo** [comment]
Hi, thanks for your feedback :+1: 

there's no team here, just me :wink: 

Although this library can emulate an usb soft host, it only supports HID 1.0 devices.

An esp32-S3 would be a much better fit for you needs.

**Reply by sdellava** [reply]
Great team anyway 👍 

Yes, it seems that is a good hw, but I still need a usb library to read and write files.

**Reply by tobozo** [reply]
[USB-MSC](https://github.com/espressif/arduino-esp32/tree/master/libraries/USB/examples/USBMSC) works with S3
